### PR TITLE
Add pipeline to run s390x triggers tests

### PR DESCRIPTION
**Pull request by barthy1**
# Changes

- add corresponding cronjob to trigger s390x tekton triggers tests
- add template to run s390x trigger test pipeline
  - k8s cluster installation
  - pipeline and triggers deployment
  - e2e triggers test
  - k8s cluster removal

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      - name: target-arch
+        value: $(tt.params.targetArch)
+      - name: remote-host
+        value: $(tt.params.remoteHost)
+      - name: remote-port
+        value: $(tt.params.remotePort)
+      - name: remote-user
+        value: $(tt.params.remoteUser)
+      resources:
+      - name: tekton-pipeline-source
```
Comment (afrittoli) [inline tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml:178]: I'd rather not continue adding new pipeline resources, but as mentioned earlier I'm ok with it for now.
I'd love to see at some point a follow up that removes them 🙏

**Review by afrittoli — COMMENTED** [review]
Since the pipeline definition is embedded in the trigger, you could probably manage to have a single trigger and a single pipeline definition in there and reduce duplication a bit... but that might become less readable on the other hand, and pipeline and trigger testing pipeline might also diverge in future, so it's probably better as it is now.

I have not tried this, but it looks ok, thank you!
/lgtm

File: tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml (hunk 1 above)
Comment (barthy1) [reply tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml:178]: yeah, I definitely remember about moving to git-clone usage. Could you please install this task to bastion-z namespace? and with the next PR I will use it for all s390x pipelines.

Comment (savitaashture) [reply tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml:178]: +1 on @afrittoli comment

File: tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml [hunk 2]
```diff
@@ -0,0 +1,201 @@
+# Copyright 2020 The Tekton Authors
```
Comment (savitaashture) [inline tekton/resources/nightly-tests/triggers-deploy-test-s390x-template.yaml:1]:
```suggestion
# Copyright 2021 The Tekton Authors
```

**Comment by barthy1** [comment]
@afrittoli @savitaashture thank you for review. I updated year in copyright, as per request. If you are ok that switch to workspace usage will be in next PR, could you please add lgtm and approve this one?

**Review by afrittoli — COMMENTED** [review]
/approve

**Comment by barthy1** [comment]
And one more question from my side. I started to work already on "workspace instead of PipelineResource" approach and faced that 2 workspaces(in my case first one for source code and second one for k8s config) with 2 PVCs accordingly is a reason of TaskRunValidationFailed failure - "more than one PersistentVolumeClaim is bound". 
What is best practice in this case, just use 1 PVC and 1 workspace for all information which needs to be shared between tasks?

**Comment by afrittoli** [comment]
> And one more question from my side. I started to work already on "workspace instead of PipelineResource" approach and faced that 2 workspaces(in my case first one for source code and second one for k8s config) with 2 PVCs accordingly is a reason of TaskRunValidationFailed failure - "more than one PersistentVolumeClaim is bound".
> What is best practice in this case, just use 1 PVC and 1 workspace for all information which needs to be shared between tasks?

Yes, I would say use one workspace at pipeline level, so you will provision only one PVC.
You can use as many workspaces as you want in your tasks, and you can pass different paths of the pipeline workspace to your task workspaces.

**Comment by savitaashture** [comment]
/lgtm

Thank you @barthy1 :+1:

**Pull request merged**